### PR TITLE
Feature/data updates

### PR DESCRIPTION
**Pull request by noahsettersten**
This rolls up a number of data model and seeding changes for Voyage:
* A new `dev:prime` task has been created that sets up additional test users, while `db:seed` has been refactored to only create the initial admin user no matter what environment you're in.
* `factory_girl` is replaced with `factory_bot` across the app.
* `paper_trail` has been replaced with `paranoia` and the User model has been set up to store deleted users by default.

This addresses issues #12, #35, #52, and #53

### CODE REVIEW

File: lib/voyage/app_builder.rb [hunk 1]
```diff
@@ -481,11 +481,11 @@ def customize_user_factory(adding_first_and_last_name)
       inject_into_file 'spec/factories/users.rb', before: /^  end/ do <<-'RUBY'.gsub(/^ {4}/, '')
         password 'asdfjkl123'
         password_confirmation 'asdfjkl123'
-        sequence(:email) { |n| "user_#{n}@example.com" }
+        email { "user_#{uuid}@example.com" }
```
Comment (GeekOnCoffee) [inline lib/voyage/app_builder.rb:484]: This will make logging in hard, as I believe we prime using these, so we get user_1@example.com

Comment (GeekOnCoffee) [reply lib/voyage/app_builder.rb:484]: Though it looks like you're manually creating an admin below, so maybe that's not a big deal.

**Review by GeekOnCoffee — APPROVED** [review]

File: lib/voyage/app_builder.rb (hunk 1 above)
Comment (noahsettersten) [reply lib/voyage/app_builder.rb:484]: I'd really rather not use the UUID at all for these factories, but the current test setup doesn't appear to properly reset the database between each test. Maybe if we integrate / setup DatabaseCleaner that would obviate any need for this.

Comment (GeekOnCoffee) [reply lib/voyage/app_builder.rb:484]: Is it hitting an issue creating a user again? I would have thought the sequence would have been consecutive over the test run

Comment (noahsettersten) [reply lib/voyage/app_builder.rb:484]: I would have thought so too, but something else appears to be at play here. Once we button up these few remaining issues, I hope to go back and review the specs to get everything passing successfully and bump the coverage if there's time as well.

**Pull request merged**